lib/elixir/lib/module/types.ex

```diff
@@ -464,14 +464,16 @@ defmodule Module.Types do
   ## Module errors
 
   def format_error({:unused_args, {name, arity}}),
-    do: "default values for the optional arguments in #{name}/#{arity} are never used"
+    do:
+      "default values for the optional arguments in the private function #{name}/#{arity} are never used, you can make this function public to silence this warning"
```

Comment:
I don’t think the suggestion is necessarily correct. It may be the best suggestion is to remove the default argument.

Reply:
Perhaps we should say: “you may remove it”/“you may remove them”?

Reply:
If you prefer I will do as you suggest, but FWIW, I encountered this when I changed a public function with a default into a private function with a default that was only used in certain build conditions (MIX_ENV=dev versus MIX_ENV=test type thing).  The error was only produced in the build I happened to be using at the time. Removing them would have broken the other builds.  Honestly what I really wanted was a way to silence the warning entirely, but I couldn't find any prior art to crib from. I ended making the function public to avoid it.

Maybe its best to just remove the suggestion on what to do, and leave the new reference to this only affecting private functions. That was what my colleagues found most confusing about this error when I asked about it.

Reply:
Updated to remove the suggestion.